include/wx/dcsvg.h

```diff
@@ -323,7 +323,6 @@ class WXDLLIMPEXP_CORE wxSVGFileDC : public wxDC
 
     void SetShapeRenderingMode(wxSVGShapeRenderingMode renderingMode);
 
-private:
```

Comment:
Hmm, I don't really like this.

This ctor was originally added in 26b273bef91 to make wxRTTI work and I'm not even sure if adding it was a good idea: perhaps it would have been better to use `wxDECLARE_ABSTRACT_CLASS` for this class instead. But as long as this ctor was private, it was still acceptable.

Making it public is IMO not a good idea and it looks like it was done in 80053d014b just to fix the build with XTI which we plan to remove anyhow (I spoke about doing it many times and will sooner or later finally do it...). So I'd like to revert this part of 80053d014b (at least some other changes of this commit are definitely correct, so I wouldn't want to revert it entirely) and/or make this class "abstract" from wxRTTI point of view.

What do you think?

Reply:
If XTI will be removed, this is indeed not necessary.

I intentionally used  `wxDECLARE_DYNAMIC_CLASS` for `wxSVGFileDC` so you can just (try to) cast  a `wxDC` to `wxSVGFileDC` and use it specific features (like `SetShapeRenderingMode`).

For example, in the drawing sample you can add to `MyCanvas::Draw` the following code to use crisp edges when saving as SVG:
```c++
if ( wxSVGFileDC *svgDC = wxDynamicCast(&pdc, wxSVGFileDC) )
{
    svgDC->SetShapeRenderingMode(wxSVG_SHAPE_RENDERING_CRISP_EDGES);
}
```

Reply:
Just to be clear: I don't suggest removing wxRTTI for this class entirely, just making it `ABSTRACT` rather than `DYNAMIC`, i.e. not allowing to create objects of this type without any parameters.

Reply:
Ah, I must have misunderstood. That seems like a simple change. I can add it to this PR.